### DISCUSSION TITLE
EarlyStopping callback defaults

### DISCUSSION THREAD

**Discussion by sachinruk** [Lightning-AI/pytorch-lightning#19363] [code help: CV]
Just some questions about the defaults/ best practices.

IIUC looks like you should be adding _epoch to the monitor variable. Also, imo you should set check_on_train_epoch_end = False to make sure everything runs after validation set has completed. Defauly behaviour seems to be after training set completed. This will be an issue considering the validation metrics are stiil from the previous epch?
```python
early_stopping_callback = EarlyStopping(
    monitor="validation_loss_epoch",
    check_on_train_epoch_end=False,
    ...,
)
```
pytorch_lightning version: 2.1.3

## Edit
I actually realised that when `check_on_train_epoch_end=False,` it goes to `on_validation_end` and there is no `on_validation_epoch_end` in the EarlyStopping callback. According to [the diagram](https://lightning.ai/docs/pytorch/latest/common/lightning_module.html#hooks) here at least, on_validation_end == on_validation_epoch_end. Please correct me if I am wrong.
